x509-cert/src/serial_number.rs

```diff
@@ -65,6 +75,46 @@ impl<P: Profile> SerialNumber<P> {
     pub fn as_bytes(&self) -> &[u8] {
         self.inner.as_bytes()
     }
+
+    /// Generates a random serial number from RNG.
+    ///
+    /// This follows the recommendation the CAB forum [ballot 164] and uses a minimum of 64 bits
+    /// of output from the CSPRNG.
+    ///
+    /// [ballot 164]: https://cabforum.org/2016/03/31/ballot-164/
+    pub fn generate<N>(rng: &mut impl CryptoRngCore) -> Result<Self>
```

Comment:
Is there a smart way to provide a default type?

Reply:
If I were writing this I would probably do a `generate` method which does the recommended 17 bytes along with a non-const-generic parameterized `generate_with_length` which allows configuring it at runtime.

The type safe enforcement of the lengths is cool and all, but if we're going to allow overriding of a preferred default, it would probably be nice to allow it to be runtime-configurable if necessary.

Reply:
The reason I went with const is that I didn't expect people to put the length in configuration but put that statically in their code.

I could make the generate a non-parameterized version that puts 17-bytes and provide a `pub fn generate_with_length<N>(rng: &mut impl CryptoRngCore)`.